pkg/reporting/send.go

```diff
@@ -37,8 +41,10 @@ type eventSender[E any] struct {
 }
 
 func (s eventSender[E]) senderLoop(ctx context.Context, logger *zap.Logger) {
-	ticker := time.NewTicker(time.Second * time.Duration(s.client.BaseConfig.PushEverySeconds))
-	defer ticker.Stop()
+	heartbeat := time.Second * time.Duration(s.client.BaseConfig.PushEverySeconds)
+
+	timer := time.NewTimer(heartbeat)
```

Comment:
question, suggestion: Since we are effectively have a loop here, should it be ticker instead of timer?